### ISSUE TITLE
analysis: NDVI・NDWI の多重共線性への対処方針を決める（Limited シナリオ）

### ISSUE THREAD

**Issue by HitsujiNeko** [HitsujiNeko/MasterResearch#268]
## 概要

`NDVI` と `NDWI` の強い共線性（Pearson -0.966、VIF 16.8〜27.0）への対処方針を、指数の差し替えを含む4構成の比較によって決定する。判断は `Limited` シナリオで行う。

## 背景・目的

`NDVI` × `NDWI` の Pearson は **-0.966** であり、両変数の VIF は全ランで **16.8〜27.0**（危険水準 VIF > 10 を大きく超過）である。`Satellite Only` でも 21.09・20.05 と同水準であり、シナリオを問わず継続している。

### 原因は指数の定義そのものにある

本研究の `NDWI` は McFeeters 型（`(GREEN - NIR) / (GREEN + NIR)`）である。`NDVI` と並べると、両者が同じバンドに支配されていることが分かる。

| 指数 | Blue | Green | Red | NIR | SWIR1 | SWIR2 |
|---|:---:|:---:|:---:|:---:|:---:|:---:|
| `NDVI` | | | - | **+** | | |
| `NDWI`（現行） | | + | | **-** | | |
| MNDWI | | + | | | - | |
| ANDWI | + | + | + | - | - | - |

`NDVI` と `NDWI` は**どちらも NIR に支配され、符号が逆**である。植生は NIR 反射が非常に強く、水域はほぼ無いため、`NDVI` が上がれば `NDWI` は必ず下がる。**-0.966 はハノイの土地被覆に固有の事象ではなく、指数の定義から生じる構造的な相関である。**

### 両方を残す限り VIF は 15 を下回れない

2変数だけで計算した VIF の**下界**は次のとおりである。

```text
1 / (1 - 0.966^2) = 14.96
```

他の変数が何本あっても VIF はこの値を下回れないため、**`NDVI` と `NDWI` を同時に投入する限り危険水準（> 10）は解消できない**。

### 現れている症状

1. **VIF が危険水準超え**: 16.8〜27.0
2. **標準化係数が解釈不能**: 5.7節は `NDVI`・`NDWI` とも係数が負（冷却方向）と記録している。しかし r = -0.966 の2変数はほぼ同一の軸を符号反転して測っており、「両方が冷却」という読みは単純には成り立たない
3. **寄与が分割されている可能性**: `NDVI` の SHAP 0.5215（2位）、`NDWI` 0.3131（3位）。本来1つの情報が2変数に割れているなら SHAP 順位も歪む

RQ1（どの説明変数が LST に対して支配的か）に答えるうえで、この状態の解消可否を判断する必要がある。

### 共線性を診断した先行研究は登録文献に無い

| 論文 | `NDVI`・`NDWI` の扱い | 共線性の診断 |
|---|---|---|
| S3 Onačillová 2022 | NDVI/NDBI/NDWI を MLR に同時投入 | なし |
| S6 Garzón 2021 | NDVI/NDBI/NDWI＋距離。NDWI の寄与率 51.46% が最大 | 冗長ペア（NDVI と Greenness、R²=0.99）の一方を除外する変数選択は実施。ただし NDVI と NDWI は両方残す |
| S10 Hoang 2025 | NDVI/NDBI/ANDWI/NDBSI を LightGBM+SHAP へ投入 | なし（PDF 全文に `multicollinear` / `collinear` の語が1件も無いことを確認） |

**`NDVI` と `NDWI` を同時投入したうえで共線性を診断した研究は、登録文献の中に無い。** 本研究が VIF と相関を報告すること自体が上積みになる。

### 指数の使用頻度（S7 Derdouri 2021・SUHI 研究113本のレビュー）

Table A2 より `NDWI` 9.8%、**MNDWI 8.5%**、LSWI 1.2%、NDMI 1.2%。水指数では `NDWI` と MNDWI が2大選択肢であり、**MNDWI への差し替えは一般的手法の範囲に収まる**。

ただし表の `NDWI` 9.8% は参考文献 [184] Gao 1996（`(NIR - SWIR)/(NIR + SWIR)`・植生水分）と [185] McFeeters 1996（`(GREEN - NIR)/(GREEN + NIR)`・開水面）の**2つの異なる定義を合算した値**である。本研究が使うのは McFeeters 型のみであり、その単独の使用率は原本から分離できない（未確認）。

なお S7 は本文と Table A2 で数値が異なる（本文は `NDWI` 7.3%・MNDWI 6.4%・`NDVI` 52.5%・`NDBI` 24.5%）。差の理由は原本から特定できていない。**引用時はどちらの値か明記する。**

### MNDWI の技術的根拠

MNDWI の出典は Xu, H. (2006) *Int. J. Remote Sens.* 27, 3025-3033 であり、題名のとおり **NDWI の弱点を修正する目的で提案された指標**である。

- **NDWI（McFeeters 型）は市街地に反応してしまい、水域を過大評価する**
- **MNDWI は NIR を SWIR に置き換えることで市街地のノイズを抑制する**
- トレードオフ: 植生の影を水と誤分類する場合がある

この「市街地ノイズの抑制」は本研究にとって副次的な利点でもある。ハノイのような高密度市街地では、現行の `NDWI` が市街地を水域と誤認している可能性がある。

### 類似研究例

- **Zhu et al. (2023), *Ecological Indicators*, 厦門島（中国）**: `NDVI`・`NDBI`・**MNDWI**・建物面積の2D指標に3D形態5因子を加え、季節別 LST を街区スケールで分析。**`NDVI` と MNDWI を同時投入した直接の前例**である。多重共線性・VIF の議論は無い。島嶼都市では秋冬に MNDWI と LST が正相関（水の比熱が高いため）という季節性の報告がある
- **Hyderabad（インド）**: RF・XGBoost による LST 予測で MNDWI が LST と最も強い正相関（r = 0.49）
- **Zonguldak（トルコ）**: `NDVI`・SAVI・MNDWI・`NDBI` の相関分析＋MLR で将来 LST を予測

Zhu et al. (2023) は文献データベースへの登録が別 Issue で起票済みであり、**登録が済めば本 Issue の根拠として直接引用できる**。

上記のうち Zhu et al. (2023) 以外は検索結果の要約に基づくものであり、**原本PDFでの確認は未実施**である。

### ベトナム都市での前例は確認できていない

ハノイ・ベトナムの LST 研究を検索した範囲では `NDVI`・`NDBI`・`NDWI`・Bare Soil Index の組み合わせが主流であり、MNDWI を使う例は見つからなかった（無いことの証明ではない）。

### 本 Issue の位置づけ

**「MNDWI を使うこと」自体は既存手法の踏襲であり、「共線性を診断したうえで指数を選ぶこと」が本研究の独自部分である。** この線引きを結果ドキュメントに明記する。

### 比較する4構成

| 構成 | 内容 | 位置づけ |
|---|---|---|
| A ベースライン | 現行（`NDVI` + `NDWI`） | 比較の基準 |
| B 一方を除外 | `NDWI` を除外し `NDVI` のみ | S6 に前例あり。SHAP 3位の寄与を失う |
| C 主成分化 | 2変数を第1主成分に統合 | VIF は消えるが物理的解釈ができない |
| D 指数の差し替え | `NDWI` → **MNDWI** | 共線性の原因（NIR 共有）を除去しつつ水域の情報を保持する |

A〜C は「変数を減らす」または「現状を追認する」対処であるのに対し、**D のみが原因そのものを取り除く**。MNDWI は NIR を使わないため、定義上 `NDVI` との相関は下がることが期待される。

### 実装方針（案）

`gee_calc_satellite_indices.py` の `add_indices()` に MNDWI を追加する。

```text
MNDWI = (GREEN - SWIR1) / (GREEN + SWIR1)
```

`get_scaled_optical_bands()` が既に取得しているバンドは `SR_B3`（green）・`SR_B4`（red）・`SR_B5`（NIR）・`SR_B6`（SWIR1）であり、**MNDWI に必要な GREEN・SWIR1 は取得済みである。バンド選択の変更は不要で、式を1本追加するだけで算出できる。**

ANDWI（S10 が採用）を候補に加える場合は `SR_B2`（blue）と `SR_B7`（SWIR2）の追加が必要になる。本 Issue では MNDWI を主候補とし、ANDWI は MNDWI で共線性が十分下がらなかった場合の次善策として扱う。

## 条件

- **判断は `Limited` シナリオで行う**。MNDWI はデータセット共通の列となるため `Satellite Only` でも利用可能になるが、そちらへの反映は本 Issue では行わず後続 Issue とする。**その間シナリオ間で変数定義が食い違う状態になることを結果ドキュメントに明記する**
- 4構成は**同一セル上で**評価する。フィルタの非NULL要求列は構成によらず固定し、母数を揃える
- 建物高さの構成は本 Issue で再検討せず、その時点で採用されている構成をそのまま用いる
- 既存結果との比較可能性を保つため、観測日・スケール・グリッド定義・乱数シードは現行の主結果と揃える
- 主成分化を行う場合、Spatial CV では fold 内で fit することを検討する
- **MNDWI で共線性が十分下がる保証はない**。SWIR1 も土地被覆と相関するため相関が残る可能性があり、その場合は B または A を採用する判断もありうる。結論を先取りしない
- 結果ドキュメントは `/report-results` の枠組みに沿って更新する

## 完了条件

- [ ] MNDWI の算出が実装され、テストが通過している
- [ ] MNDWI を含むデータセットが再生成され、`NDVI` × MNDWI の相関が実測されている
- [ ] 4構成（A ベースライン／B `NDWI` 除外／C 主成分化／D MNDWI 差し替え）が**同一セル上で**実行され、結果ファイルが出力されている
- [ ] 各構成の VIF・相関・Spatial CV R²（RF・線形）・RF 重要度・SHAP が記録されている
- [ ] 水域の情報が保持されているか（MNDWI の SHAP 寄与が `NDWI` から大きく落ちていないか）が確認されている
- [ ] 採用する構成が根拠つきで決定され、以後の `Limited` 分析の既定構成として反映されている
- [ ] 採用後に線形回帰の標準化係数を主たる解釈に用いてよいかの判断が記録されている
- [ ] 5.7節の符号の解釈（`NDVI`・`NDWI` がともに負）について、採用構成での状況が再確認されている
- [ ] 先行研究との位置づけ（MNDWI の使用自体は踏襲・共線性診断を伴う選択が独自）が結果ドキュメントに記録されている
- [ ] `Satellite Only` との変数定義の不整合が残ることと、その解消を後続 Issue で行う旨が明記されている
- [ ] `docs/03_results/limited_analysis_results.md`（5.1節・5.7節・6.2節・6.5節・7節）が更新され、`docs/README.md` が更新されている

## 関連ファイル

実装:
- `src/gee/gee_calc_satellite_indices.py`（`add_indices()`・`get_scaled_optical_bands()`）
- `src/analysis/analysis_rq3_limited.py`（`SPECTRAL_FEATURE_COLUMNS`・`resolve_feature_columns()`）
- `src/analysis/urban_params/config.py`（`RASTER_KEYS`）
- `src/analysis/calc_urban_params.py`（`RASTER_KEYS`）
- `src/common/model_metrics.py`（`compute_vif`）

テスト:
- `tests/analysis/test_analysis_rq3_limited.py`
- `tests/common/test_model_metrics.py`

ドキュメント:
- `docs/03_results/limited_analysis_results.md`（5.1節・5.7節・6.2節・6.5節・7節）
- `docs/03_results/satellite_only_analysis_results_cellbased.md`（5.3節・不整合の明記先）
- `docs/01_planning/urban_structure_parameters.md`（P18 NDWI の定義・採用根拠）
- `docs/02_methods/analysis_rq3_satellite_only_guide.md`（VIF の算出・解釈）

## 関連 Issue

- 関連: #207（Zhu et al. 2023 の文献データベース登録。本 Issue の根拠として引用したい）
- 関連: #266（建物高さ列の多重共線性。同じ VIF 問題だが原因も対処法も異なる）

## 参考

- 起票時点の VIF・相関の実測値: `docs/03_results/limited_analysis_results.md` 6.1節・6.2節・6.5節
- MNDWI の出典: Xu, H. Modification of Normalised Difference Water Index (NDWI) to Enhance Open Water Features in Remotely Sensed Imagery. *Int. J. Remote Sens.* 2006, 27, 3025-3033
- ANDWI の出典: Rad, A.M.; Kreitler, J.; Sadegh, M. Augmented Normalized Difference Water Index for improved surface water monitoring. *Environ. Model. Softw.* 2021, 140, 105030
- 類似研究: Zhu, Z. et al. How does 2D and 3D of urban morphology affect the seasonal land surface temperature in Island City? A block-scale perspective. *Ecological Indicators* 2023
